treeherder/etl/resultset_loader.py

```diff
@@ -86,34 +86,31 @@ def fetch_resultset(self, url, repository, sha=None):
         params.update(self.CREDENTIALS)
 
         logger.info("Fetching resultset details: {}".format(url))
-        try:
-            commits = self.get_cleaned_commits(fetch_json(url, params))
-            head_commit = commits[-1]
-            resultset = {
-                "revision": head_commit["sha"],
-                "push_timestamp": to_timestamp(
-                    head_commit["commit"]["author"]["date"]),
-                "author": head_commit["commit"]["author"]["email"],
-            }
-
-            revisions = []
-            for commit in commits:
-                revisions.append({
-                    "comment": commit["commit"]["message"],
-                    "author": "{} <{}>".format(
-                        commit["commit"]["author"]["name"],
-                        commit["commit"]["author"]["email"]),
-                    "revision": commit["sha"]
-                })
-
-            resultset["revisions"] = revisions
-            return resultset
-
-        except Exception as ex:
-            logger.exception("Error fetching commits", exc_info=ex)
-            newrelic.agent.record_exception(ex, params={
-                "url": url, "repository": repository, "sha": sha
-                })
+        newrelic.agent.add_custom_parameter("url", url)
+        newrelic.agent.add_custom_parameter("repository", repository)
+        newrelic.agent.add_custom_parameter("sha", sha)
```

Comment:
Should we put these in ResultsetLoader.process? (or at least all but the SHA one, since I don't think we have sha then)

Doing so would mean the annotations are set sooner (in case of exceptions earlier in the codepath), plus means a future Hg transformer would have to duplicate less.